plugin/storage/es/factory.go

```diff
@@ -190,32 +194,69 @@ func createSpanWriter(
 }
 
 // GetSpanServiceMappings returns span and service mappings
-func GetSpanServiceMappings(shards, replicas int64, esVersion uint) (string, string) {
+func GetSpanServiceMappings(shards, replicas int64, esVersion uint, esPrefix string, useILM bool) (string, string, error) {
 	if esVersion == 7 {
-		return fixMapping(loadMapping("/jaeger-span-7.json"), shards, replicas),
-			fixMapping(loadMapping("/jaeger-service-7.json"), shards, replicas)
+		spanMapping, err := FixMapping(newTextTemplateBuilder(), LoadMapping("/jaeger-span-7.json"), shards, replicas, esPrefix, useILM)
+		if err != nil {
+			return "", "", err
+		}
+		serviceMapping, err := FixMapping(newTextTemplateBuilder(), LoadMapping("/jaeger-service-7.json"), shards, replicas, esPrefix, useILM)
+		if err != nil {
+			return "", "", err
+		}
+		return spanMapping, serviceMapping, nil
+	}
+	spanMapping, err := FixMapping(newTextTemplateBuilder(), LoadMapping("/jaeger-span.json"), shards, replicas, "", false)
+	if err != nil {
+		return "", "", err
+	}
+	serviceMapping, err := FixMapping(newTextTemplateBuilder(), LoadMapping("/jaeger-service.json"), shards, replicas, "", false)
+	if err != nil {
+		return "", "", err
 	}
-	return fixMapping(loadMapping("/jaeger-span.json"), shards, replicas),
-		fixMapping(loadMapping("/jaeger-service.json"), shards, replicas)
+	return spanMapping, serviceMapping, nil
 }
 
 // GetDependenciesMappings returns dependencies mappings
-func GetDependenciesMappings(shards, replicas int64, esVersion uint) string {
+func GetDependenciesMappings(shards, replicas int64, esVersion uint) (string, error) {
 	if esVersion == 7 {
-		return fixMapping(loadMapping("/jaeger-dependencies-7.json"), shards, replicas)
+		return FixMapping(newTextTemplateBuilder(), LoadMapping("/jaeger-dependencies-7.json"), shards, replicas, "", false)
 	}
-	return fixMapping(loadMapping("/jaeger-dependencies.json"), shards, replicas)
+	return FixMapping(newTextTemplateBuilder(), LoadMapping("/jaeger-dependencies.json"), shards, replicas, "", false)
 }
 
-func loadMapping(name string) string {
+// LoadMapping returns index mappings from go assets as strings
+func LoadMapping(name string) string {
 	s, _ := mappings.FSString(false, name)
 	return s
 }
 
-func fixMapping(mapping string, shards, replicas int64) string {
-	mapping = strings.Replace(mapping, "${__NUMBER_OF_SHARDS__}", strconv.FormatInt(shards, 10), 1)
-	mapping = strings.Replace(mapping, "${__NUMBER_OF_REPLICAS__}", strconv.FormatInt(replicas, 10), 1)
-	return mapping
+// FixMapping parses the index mappings with given values and returns parsed template as string
+func FixMapping(tb es.TemplateBuilder, mapping string, shards, replicas int64, esPrefix string, useILM bool) (string, error) {
+
+	tmpl, err := tb.Parse(mapping)
+	writer := new(bytes.Buffer)
```

Comment:
nit: this can go after the `err != nil` check.

Reply:
my bad, would be fixed in next commit